# HG changeset patch
# User Rik <rik@octave.org>
# Date 1356797704 28800
#      Sat Dec 29 08:15:04 2012 -0800
# Node ID c1c6502fe52bee565a58f776885ab80bc9b5ccc7
# Parent  500650d6ddf5a54be7031910c7f070627889aca3
oct-rand.cc: Silence compiler warning about possible uninitialized variable.

* liboctave/numeric/oct-rand.cc(do_float_scalar): initialize dretval
to stop compiler warning.

diff --git a/liboctave/numeric/oct-rand.cc b/liboctave/numeric/oct-rand.cc
--- a/liboctave/numeric/oct-rand.cc
+++ b/liboctave/numeric/oct-rand.cc
@@ -422,17 +422,17 @@ octave_rand::do_scalar (double a)
 float
 octave_rand::do_float_scalar (float a)
 {
   float retval = 0.0;
 
   if (use_old_generators)
     {
       double da = a;
-      double dretval;
+      double dretval = 0.0;
       switch (current_distribution)
         {
         case uniform_dist:
           F77_FUNC (dgenunf, DGENUNF) (0.0, 1.0, dretval);
           break;
 
         case normal_dist:
           F77_FUNC (dgennor, DGENNOR) (0.0, 1.0, dretval);
